### ISSUE TITLE
regalloc2 upgrade being pulled in somehow with older versions of wasmtime crate 

### ISSUE THREAD

**Issue by gamemachine** [bytecodealliance/wasmtime#9192]
Did a cargo update today with no other changes.  wasmtime crate is v15.  No other bytecodealliance crates.  Build failed on cranelift vscode because something is pulling in a newer version of regalloc2.  Is something in wasmtime just pulling in the most recent of something? Just guessing.

**Comment by alexcrichton** [comment]
cc @cfallin  and @elliottt, could y'all yank 0.9.4 of regalloc2 and republish it as 0.10.0? Locally I see:

```
$ cargo install wasmtime-cli@15.0.0
...
error[E0599]: no function or associated item named `forward` found for struct `InstRange` in the current scope
    --> /home/alex/.cargo/registry/src/index.crates.io-6f17d22bba15001f/cranelift-codegen-0.102.1/src/machinst/vcode.rs:1340:20
     |
1340 |         InstRange::forward(start, end)
     |                    ^^^^^^^ function or associated item not found in `InstRange`
     |
note: if you're trying to build a new `InstRange`, consider using `InstRange::new` which returns `InstRange`
    --> /home/alex/.cargo/registry/src/index.crates.io-6f17d22bba15001f/regalloc2-0.9.4/src/index.rs:148:5
     |
148  |     pub fn new(from: Inst, to: Inst) -> Self {
     |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
...
```

and it looks like this method was removed in 0.9.4

**Comment by elliottt** [comment]
Will do, sorry about the trouble!

**Comment by elliottt** [comment]
I've yanked 0.9.4, and will release 0.10.0 on Tuesday. I'll update wasmtime at the same time :+1:

**Comment by alexcrichton** [comment]
With https://github.com/bytecodealliance/wasmtime/pull/9193 I'm going to consider this fixed (thanks @elliottt!) and the 0.10.0 upgrade will likely happen today but can be tracked independently of this.
